core/tests/ts-integration/tests/api/web3.test.ts

```diff
@@ -808,6 +808,18 @@ describe('web3 API compatibility tests', () => {
         ).toBeRejected();
     });
 
+    test('Should check v value of the legacy transaction', async () => {
+        const LEGACY_TX_TYPE = 0;
+        let tx = await alice.sendTransaction({
+            type: LEGACY_TX_TYPE,
+            to: alice.address
+        });
+        let receipt = await tx.wait();
+
+        let block_with_transactions = await alice.provider.send('eth_getBlockByHash', [receipt.blockHash, true]);
+        expect(block_with_transactions.transactions[0].v >= 27);
```

Comment:
You shouldn't rely on the fact that the transaction is the first in the block. You can just use `provider.getTransaction`. Nit: variables name should be camelCase -- `blockWithTransactions`, etc